src/connect-proxy/worker.ts

```diff
@@ -32,25 +35,40 @@ import * as device from './device';
 		}
 	});
 
-const VPN_SERVICE_API_KEY = process.env.VPN_SERVICE_API_KEY!;
+const VPN_SERVICE_API_KEY = Buffer.from(process.env.VPN_SERVICE_API_KEY!);
 
-const tunnelToDevice: Middleware = (req, cltSocket, _head, next) =>
-	Promise.try(() => {
-		if (req.url == null) {
-			throw new Error('Bad Request');
-		}
+const lookupAsync = Promise.promisify(dns.lookup);
 
-		const match = req.url.match(
-			/^([a-fA-F0-9]+).(?:resin|balena)(?::([0-9]+))?$/,
-		);
-		if (match == null) {
-			throw new Error(`Invalid hostname: ${req.url}`);
-		}
-		const [, uuid, port = '80'] = match;
-		Raven.setContext({ user: { uuid } });
-		logger.info(`tunnel requested for ${uuid}:${port}`);
+const parseRequest = (req: nodeTunnel.Request) => {
+	if (req.url == null) {
+		throw new Error('Bad Request');
+	}
+
+	const match = req.url.match(
+		/^([a-fA-F0-9]+)\.(balena|resin|vpn)(?::([0-9]+))?$/,
+	);
+	if (match == null) {
+		throw new Error(`Invalid hostname: ${req.url}`);
+	}
+	const [, uuid, tld, port = '80'] = match;
+	if (tld === 'resin') {
+		logger.warn(`'.resin' tld is deprecated, use '.balena'`);
+	}
+
+	let auth = undefined;
+	if (req.auth != null && req.auth.password != null) {
+		auth = Buffer.from(req.auth.password);
```

Comment:
Sometimes it needs to be base64 encoded, sometimes it doesn't. I find with a string it can get confusing as to what encoding you're being passed but with a Buffer you know exactly where you stand.